src/services/uploader/uploading.controller.ts

```diff
@@ -88,27 +91,28 @@ export class UploadingController {
   }
 
   @UseGuards(AuthGuard('jwt'))
+  @UseInterceptors(UserDetailsInterceptor)
   @Post('start_encode')
   async startEncode(@Body() body: StartEncodeDto, @Request() req) {
     const request = parseAndValidateRequest(req, this.#logger);
-    if (request.user.network !== 'hive') {
-      throw new HttpException(
-        'Must be signed in with a hive account to perform this operation',
-        HttpStatus.FORBIDDEN,
-      );
+    const hiveUsername: string = body.username;
```

Comment:
Might need some formatting and a little rework for type compliance, but this is the idea of [this comment]()
```suggestion
    const hiveUsername = body.username || (request.user.network === 'hive' && request.user.username);
```

Reply:
Edit: fixed link in comment